code/modules/mob/abstract/ghost/storyteller/storyteller_verbs.dm

```diff
@@ -0,0 +1,346 @@
+
+/mob/abstract/ghost/storyteller/verb/change_mob_name(var/mob/victim)
+	set name = "Change Mob Name"
+
+	var/new_name = tgui_input_text(src, "Enter a new name.", "Change Mob Name", max_length = MAX_NAME_LEN)
+	if(!new_name)
+		return
+
+	log_admin("[key_name(src)] has renamed [victim] to [new_name].")
+	victim.set_name(new_name)
+
+/mob/abstract/ghost/storyteller/verb/change_obj_name(var/obj/thing)
+	set name = "Change Object Name"
+
+	var/new_name = tgui_input_text(src, "Enter a new name.", "Change Object Name", max_length = MAX_NAME_LEN)
+	if(!new_name)
+		return
+
+	log_admin("[key_name(src)] has renamed [thing] to [new_name].")
+	thing.name = new_name
+
+/mob/abstract/ghost/storyteller/verb/change_obj_desc(var/obj/thing)
+	set name = "Change Object Description"
+
+	var/new_desc = tgui_input_text(src, "Enter a new description.", "Change Object Description", max_length = MAX_MESSAGE_LEN)
+	if(!new_desc)
+		return
+
+	log_admin("[key_name(src)] has changed [thing]'s description to [new_desc].")
+	thing.desc = new_desc
+
+/mob/abstract/ghost/storyteller/verb/set_outfit(var/mob/living/carbon/human/H)
+	set name = "Set Outfit"
+	set category = "Storyteller"
+
+	do_dressing(H)
+
+/mob/abstract/ghost/storyteller/proc/show_traitor_panel(var/mob/M)
+	set name = "Edit Antagonist"
+	set category = "Storyteller"
+
+	if(!istype(M))
+		to_chat(usr, SPAN_WARNING("This can only be used on mobs!"))
+		return
+
+	if(!M.mind)
+		to_chat(usr, SPAN_WARNING("This mob has no mind!"))
+		return
+
+	M.mind.edit_memory()
+
+/mob/abstract/ghost/storyteller/verb/create_explosion()
+	set name = "Create Explosion"
+	set category = "Storyteller"
+
+	var/turf/epicenter = loc
+	var/list/choices = list("Small Bomb", "Medium Bomb", "Big Bomb", "Custom Bomb")
+	var/choice = tgui_input_list(usr, "What size explosion would you like to produce?", "Drop Bomb", choices)
+	switch(choice)
+		if(null)
+			return 0
+		if("Small Bomb")
+			explosion(epicenter, 1, 2, 3, 3)
+		if("Medium Bomb")
+			explosion(epicenter, 2, 3, 4, 4)
+		if("Big Bomb")
+			explosion(epicenter, 3, 5, 7, 5)
+		if("Custom Bomb")
+			var/devastation_range = tgui_input_number(usr, "Set the devastation range (in tiles).", "Devastation")
+			var/heavy_impact_range = tgui_input_number(usr, "Set the heavy impact range (in tiles).", "Heavy")
+			var/light_impact_range = tgui_input_number(usr, "Set the light impact range (in tiles).", "Light")
+			var/flash_range = tgui_input_number(usr, "Set the flash range (in tiles).", "Flash")
+			explosion(epicenter, devastation_range, heavy_impact_range, light_impact_range, flash_range)
+	message_admins(SPAN_NOTICE("[ckey] creating an explosion at [epicenter.loc]."))
+
+/mob/abstract/ghost/storyteller/verb/delete_atom(atom/O as obj|mob|turf in range(world.view))
+	set name = "Delete"
+	set category = "Storyteller"
+
+	var/action = alert(src, "Are you sure you want to delete:\n[O]\nat ([O.x], [O.y], [O.z])?", "Confirmation", "Yes", "No", "Hard Delete")
+
+	if (action == "No")
```

Comment:
This is a common mistake; if the user closes the prompt, the action will be `null` and the proc would continue as if you pressed yes, which usually isn't what you want to do if you're closing it